### PR TITLE
Oceanwater 722 detect cam fault

### PR DESCRIPTION
**Pull request by lanseafood**
## Linked Issues:
| EPIC ⚡| [OCEANWATER-718](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-718) |
| :----------- | :----------- |
| Jira Ticket 🎟️   | [OCEANWATER-722](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-722) |
| Github :octocat:  | #140 #141  |

please ignore commented out code, there will be a larger cleanup PR once all things are merged to #140 

## Summary of Changes
* moved cam fault detection to new node handle
* timer compares the original vs new topic

## Test
* see tests from #129

### CODE REVIEW

**Comment by kmdalal** [comment]
I'm taking myself off this review (I think one review is enough) and prefer to review the Epic when it's ready.

**Comment by lanseafood** [comment]
@Samahu ready for review

**Review by scheidere — COMMENTED** [review]
Since I am not super familiar with the code, I focused on testing. I ran the remapping and camera faults tests easily. I am having issues tracking fault statuses for the subsequent tests. I ran "roslaunch ow atacama_y1a.launch" and I can trigger cam faults using "rosrun dynamic_reconfigure dynparam set /faults camera_left_trigger_failure true" but running "rostopic echo /faults/cam_faults_status" results in a warning that that topic has not been published yet. I am not sure what I am missing, but I would be happy to test again once that becomes clear!

**Comment by lanseafood** [comment]
> Since I am not super familiar with the code, I focused on testing. I ran the remapping and camera faults tests easily. I am having issues tracking fault statuses for the subsequent tests. I ran "roslaunch ow atacama_y1a.launch" and I can trigger cam faults using "rosrun dynamic_reconfigure dynparam set /faults camera_left_trigger_failure true" but running "rostopic echo /faults/cam_faults_status" results in a warning that that topic has not been published yet. I am not sure what I am missing, but I would be happy to test again once that becomes clear!

Not sure what the issue is, as I just reran and looks like the rostopic echo is working for me. Maybe try pulling again and seeing if you can see the topic. @scheidere

**Comment by Samahu** [comment]
> Since I am not super familiar with the code, I focused on testing. I ran the remapping and camera faults tests easily. I am having issues tracking fault statuses for the subsequent tests. I ran "roslaunch ow atacama_y1a.launch" and I can trigger cam faults using "rosrun dynamic_reconfigure dynparam set /faults camera_left_trigger_failure true" but running "rostopic echo /faults/cam_faults_status" results in a warning that that topic has not been published yet. I am not sure what I am missing, but I would be happy to test again once that becomes clear!

I don't see this issue, I am able to monitor this topic. Did you source your `devel/setup.bash` from the terminal you ran the `rostopic echo` form?

File: ow_faults_detection/src/FaultDetector.cpp [hunk 1]
```diff
@@ -8,10 +8,130 @@
...
+
 FaultDetector::FaultDetector(ros::NodeHandle& node_handle)
 {
-  
   srand (static_cast <unsigned> (time(0)));
-  m_pub = node_handle.advertise<ow_faults::SystemFaults>("/test", 10);
+
+ auto image_trigger_str = "/StereoCamera/left/image_trigger";
+  m_camera_original_trigger_sub = node_handle.subscribe(string("/_original") + image_trigger_str,
+    10, &FaultDetector::cameraTriggerOriginalCb, this);
```
Comment (Samahu) [inline ow_faults_detection/src/FaultDetector.cpp:26]: I didn't expect the FaultDetector node to make any references to the **_original** topic(s).

**Review by Samahu — CHANGES_REQUESTED** [review]
I didn't expect the FaultDetector node to make any references to the `_original/**` topic(s). This breaks the purpose of the refactoring.

**Comment by Samahu** [comment]
> I didn't expect the FaultDetector node to make any references to the `_original/**` topic(s). This breaks the purpose of the refactoring.

I do realize the reason you subscribed to the `/_original/**` but this caused me to think it wasn't probably the best way to implement camera fault this way. Rather than remapping the  `/StereoCamera/left/image_trigger` we better remap the `/StereoCamera/left/image_raw` topic.

**Comment by kmdalal** [comment]
Was curious to see what up here, and I just noticed that this is in an entirely new (?) _package_ -- I had thought there would only be a new node in the ow_faults package.  The README files in both packages are identical.  @lanseafood could you please update the README.md in this package to reflect its purpose?   (unless it is updated in the target branch)

I'd like to clearly understand the difference between fault detection at this level, and fault detection in autonomy.

**Review by scheidere — APPROVED** [review]
I figured out my issue with running the tests. It was my own confusion with some steps in rqt. Everything runs as expected for me now.

**Pull request merged**